app/service/task_status_service.rb

```diff
@@ -0,0 +1,85 @@
+class TaskStatusService
+  def initialize(form:)
+    @form = form
+  end
+
+  def mandatory_tasks_completed?
+    incomplete_tasks.empty?
+  end
+
+  def incomplete_tasks
+    { missing_pages: pages_status,
+      missing_what_happens_next: what_happens_next_status,
+      missing_privacy_policy_url: privacy_policy_status,
+      missing_contact_details: support_contact_details_status }.reject { |_k, v| v == :completed }.map { |k, _v| k }
+  end
+
+  def task_statuses
+    {
+      name_status:,
+      pages_status:,
+      declaration_status:,
+      what_happens_next_status:,
+      privacy_policy_status:,
+      support_contact_details_status:,
+      make_live_status:,
+    }
+  end
+
+private
```

Comment:
None of the statuses actually need to be accessed directly any more so can be made private